### PR TITLE
Add GA

### PR DESCRIPTION
**Pull request by axelssonHakan**

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -58,6 +58,41 @@
         "import-cwd": "2.1.0",
         "mocha": "5.2.0",
         "nyc": "13.1.0"
+      },
```
Comment (peol) [inline package-lock.json]: There's something wrong with the lock file, lots of dependencies are being downgraded/modified

File: src/index.jsx [hunk 2]
```diff
@@ -1,11 +1,32 @@
...
 import ErrorBoundary from './components/error-boundary';
 import App from './components/app';
 
 import './index.pcss';
 
+if (process.env.GA) {
+  ReactGA.initialize(process.env.GA);
+  ReactGA.pageview(window.location.pathname + window.location.search);
+
+  window.onerror = (message, source, lineno, colno, error) => {
```
Comment (peol) [inline src/index.jsx]: We should be using `addEventListener` I guess? https://developer.mozilla.org/en-US/docs/Web/API/GlobalEventHandlers/onerror#window.addEventListener('error')

File: src/index.jsx [hunk 3]
```diff
@@ -1,11 +1,32 @@
...
+    console.log('ERROR REPORTING', message, source, lineno, colno, error);
+    ReactGA.exception({
+      description: JSON.stringify(error),
+    });
+  };
+
+  window.onunhandledrejection = (error) => {
```
Comment (peol) [inline src/index.jsx]: https://developer.mozilla.org/en-US/docs/Web/Events/unhandledrejection#Simple_Reporting

File: package-lock.json (hunk 1 above)
Comment (axelssonHakan) [reply package-lock.json]: Will look into it 👍
I cerated the PR mainly to get the netlify build to validate the set env in build step. Noticed that you cant just stringfy an error so some more work to be done for this PR

Comment (axelssonHakan) [reply package-lock.json]: added a CookieConsent with similar styling as core.qlik.com

**Comment by peol** [comment]
LGTM once package-lock.json is fixed

File: src/index.jsx [hunk 4]
```diff
@@ -8,36 +8,43 @@ import App from './components/app';
...
 
-if (process.env.GA) {
-  ReactGA.initialize(process.env.GA);
-  ReactGA.pageview(window.location.pathname + window.location.search);
-
-  window.addEventListener('error', (event) => {
-    ReactGA.exception({
-      description: `ERROR: ${JSON.stringify(event, Object.getOwnPropertyNames(event))}`,
+const errorReporting = () => {
+  if (process.env.GA) {
```
Comment (peol) [inline src/index.jsx]: It hit me that we probably should use React hooks here so we unbind the event listeners in developer environments when the module gets replaced/reloaded.

Comment (axelssonHakan) [reply src/index.jsx]: Guessing the listeners will only be added if GA is set, but added useEffect(...)
Would be grateful for your review again 🙏

Comment (peol) [reply src/index.jsx]: Close! Hooks have one caveat: they *always* has to be executed in the same order during a render, so you cannot have them executed under if-clauses and the like, instead, do the if-clause inside it, or tell React to only execute it whenever specific variables change:

```js
useEffect(() => {
  // similar to "componentDidMount" or "componentDidUpdate"
  // will execute every time `process.env.GA` changes
  return () => { /* will be executed when the component gets dismounted or the condition changed for the effect (similar to "componentDidUnmount") */ };
}, [process.env.GA]);
```

**Pull request merged**